utils/keys.go

```diff
@@ -117,53 +115,47 @@ func ImportKeys(from io.Reader, to []Importer, fallbackRole string, fallbackGun
 				}
 			}
 		}
+
 		if block.Headers["gun"] == "" {
 			if fallbackGun != "" {
 				block.Headers["gun"] = fallbackGun
 			}
 		}
+
 		if block.Headers["role"] == "" {
 			if fallbackRole == "" {
 				block.Headers["role"] = notary.DefaultImportRole
 			} else {
 				block.Headers["role"] = fallbackRole
 			}
 		}
+
+		// A root key or a delegations key should not have a gun
+		// Note that a key that is not any of the canonical roles (except root) is a delegations key and should not have a gun
+		if block.Headers["role"] != tufdata.CanonicalSnapshotRole && block.Headers["role"] != tufdata.CanonicalTargetsRole && block.Headers["role"] != tufdata.CanonicalTimestampRole {
```

Comment:
Totally non-blocking nitpick:  This might be a good place for a switch, so we don't have to repeat `block.Headers["role"] !=` over and over:

```
switch block.Headers["role"] {
case tufdata.CanonicalSnapshotRole, tufdata.CanonicalTargetsRole, tufdata.CanonicalTimestampRole:
    if block.Headers...
default:
    delete(block.Headers, "gun")
}
```